datafusion/datasource/src/file_scan_config.rs

```diff
@@ -455,6 +464,10 @@ impl FileScanConfigBuilder {
             file_compression_type.unwrap_or(FileCompressionType::UNCOMPRESSED);
         let new_lines_in_values = new_lines_in_values.unwrap_or(false);
 
+        let projection = projection_indices.as_ref().map(|indices| {
+            ProjectionExprs::from_indices(indices, table_schema.table_schema())
+        });
```

Comment:
Note: this is because we are not changing `FileScanConfigBuilder`. I think it makes sense to change it at some point but it's not necessary yet and we can keep things as backwards compatible as possible for as long as possible.